src/main/java/org/apache/sling/feature/cpconverter/handlers/slinginitialcontent/VaultContentXMLContentCreator.java

```diff
@@ -0,0 +1,366 @@
+                          Set<String> removedRestrictionNames) throws RepositoryException {
+        try {
+            //set principal ACL for principal1,principal2
+            //allow jcr:read
+            StringBuilder repoInitTextSB = new StringBuilder();
+            repoInitTextSB.append("set principal ACL for " + principalId +"\n\n");
+            String path = this.currentNode.getPath();
+            if(grantedPrivilegeNames != null){
+                for(String privilege: grantedPrivilegeNames){
+                    repoInitTextSB.append("allow " + privilege + " on " + path);
+                }
+            }else if(deniedPrivilegeNames != null){
+                for(String privilege: deniedPrivilegeNames){
+                    repoInitTextSB.append("deny " + privilege + " on " + path);
+                }
+            }
+           
+
+            
+            if (((grantedPrivilegeNames != null) || (deniedPrivilegeNames != null)) && (MapUtils.isNotEmpty(restrictions) || MapUtils.isNotEmpty(mvRestrictions))) {
+              
+                if(MapUtils.isNotEmpty(mvRestrictions)){
+                   
+                    Iterator<Map.Entry<String,Value[]>> iterator = mvRestrictions.entrySet().iterator();
+                    while(iterator.hasNext()){
+                        Map.Entry<String,Value[]> entry = iterator.next();
+                        repoInitTextSB.append(appendRestrictionString(entry.getKey(), entry.getValue()));
+                    }
+                    
+                    
+                }else{
+                    Iterator<Map.Entry<String,Value>> iterator = restrictions.entrySet().iterator();
+                    while(iterator.hasNext()){
+                        Map.Entry<String,Value> entry = iterator.next();
+                        repoInitTextSB.append(appendRestrictionString(entry.getKey(), entry.getValue()));
+                    }
+                }
+                
+            }
+            repoInitTextSB.append("\nend");
+
+            repoInitTextExtensionConsumer.accept(repoInitTextSB.toString());
+        } catch (IOException | ConverterException e) {
+            throw new RepositoryException(e);
+        }
+    }
+
+    private String appendRestrictionString(String key, Value[] values){
+        Collection<String> stringValues = new ArrayList<>();
+        stringValues.add(key);
+        for(Value value: values){
+            try {
+                String valueAsString = value.getString();
+                stringValues.add(valueAsString);
+            } catch (RepositoryException e) {
+                logger.error("error appending restriction", e);
+            }
+        }
+
+        return " restriction(" +
+                stringValues.stream()
+                .map( Object::toString )
+                .collect( Collectors.joining( "," ) )
+        + ")";
+    }
+
+    private String appendRestrictionString(String key, Value value){
```

Comment:
duplicated code with appendRestrictionString(String key, Value[] values).
if you want to keep 2 methods, there is no need to build a collection of strings and joing them. you have just 1 value :)